spec/lib/publishing_api_payload/history_spec.rb

```diff
@@ -0,0 +1,110 @@
+    end
+  end
+
+  describe "#first_published_at" do
+    it "returns backdated date if edition is backdated" do
+      edition = build(:edition, first_published_at: "2020-02-22 11:00:00", backdated_to: "2020-02-20 09:00:00")
+
+      expect(described_class.new(edition).first_published_at).to eq("2020-02-20 09:00:00")
+    end
+
+    it "returns first published date if edition is not backdated" do
+      edition = build(:edition, first_published_at: "2020-02-22 11:00:00")
+
+      expect(described_class.new(edition).first_published_at).to eq("2020-02-22 11:00:00")
+    end
+  end
+
+  describe "#change_history" do
+    let(:edition) { build(:edition, first_published_at: "2020-02-22 11:00:00") }
+
+    it "appends first change note to end of history array" do
+      expected_change_note = {
+        note: "First published.",
+        public_timestamp: "2020-02-22 11:00:00",
+      }
+
+      expect(described_class.new(edition).change_history.last).to eq(expected_change_note)
+    end
+
+    it "adds change note to start of array for major changes" do
+      freeze_time do
+        second_edition = build(:edition, document: edition.document, number: 2, update_type: "major", change_note: "Some changes")
+
+        expected_change_note = {
+          note: "Some changes",
+          public_timestamp: Time.current,
+        }
+
+        expect(described_class.new(second_edition).change_history.first).to eq(expected_change_note)
+      end
+    end
+
+    it "does not add change note to start of array for minor changes" do
+      second_edition = build(:edition, document: edition.document, number: 2, update_type: "minor")
+
+      expected_change_note = {
+        note: "First published.",
+        public_timestamp: "2020-02-22 11:00:00",
+      }
+
+      expect(described_class.new(second_edition).change_history.first).to eq(expected_change_note)
+    end
+
+    it "returns change notes in reverse chronological order" do
+      change_history = [
+        {
+          note: "First update",
+          public_timestamp: "2020-02-23 09:00:00",
+        },
+        {
+          note: "Third update",
+          public_timestamp: "2020-02-25 21:00:00",
+        },
+        {
+          note: "Second update",
+          public_timestamp: "2020-02-24 12:00:00",
+        },
+      ]
+
+      second_edition = build(:edition, document: edition.document, change_history: change_history)
+
+      expected_change_notes = [
+        {
+          note: "Third update",
+          public_timestamp: "2020-02-25 21:00:00",
+        },
+        {
+          note: "Second update",
+          public_timestamp: "2020-02-24 12:00:00",
+        },
+        {
+          note: "First update",
+          public_timestamp: "2020-02-23 09:00:00",
+        },
+        {
+          note: "First published.",
+          public_timestamp: "2020-02-22 11:00:00",
+        },
+      ]
+
+      expect(described_class.new(second_edition).change_history).to eq(expected_change_notes)
+    end
```

Comment:
Be useful to also have a test that change notes set before a backdating aren't shown.

Reply:
Added a test for this